projects/projen.io/js/package.yml

```diff
@@ -0,0 +1,38 @@
+distributable:
+  url: https://github.com/projen/projen/archive/refs/tags/v{{version.raw}}.tar.gz
+  strip-components: 1
+
+versions:
+  github: projen/projen
+  strip: /^v/
+
+dependencies:
+  nodejs.org: ^22
+  classic.yarnpkg.com: ^1
+  git-scm.org: "*"
+
+build:
+  script:
+    - mkdir -p "{{prefix}}"/bin
+    - yarn version --new-version {{version.raw}} --no-git-tag-version
+    - yarn install --check-files --frozen-lockfile
+    - yarn package:js
+    - tar xfz dist/js/*.tgz -C {{prefix}}
+    - cp -R node_modules/constructs "{{prefix}}"/package/node_modules # peer dependency
+    - run: ln -s ../package/bin/projen projen
+      working-directory: ${{prefix}}/bin
+
+provides:
+  - bin/projen
```

Comment:
should the package just be `projen.io`? rather than `projen.io/js`?

Reply:
This is the tricky question i think. Projen can manage various projects next to JS/TS, e.g. Java or Go. For these projects it needs the respective language as runtime dependency. I feel like this would clutter the runtime dependencies if one only needs to use Projen on e.g. JS/TS projects.

Reply:
Hm. But it's the same binary? I'd do it in one of two ways, then:

- all one package, all feasible companions
- all one package, no companions (user includes as needed)

Reply:
Thanks for the valuable input. I'll rename the package to projen.io and change all dependencies but nodejs.org itself to companions.